### PR TITLE
Feature mes 3519 improve errors to retry integration tests

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES-3519 Refactoring autosave integration tests for clarity, and for improved testing and to ensure they are re-runnable.

Additionally improve non autosave integration tests to be runnable.

### CODE REVIEW

File: src/functions/retryProcessing/application/__tests__/common/TestEnums.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+export enum ErrorsToAbortTestCases {
+TarsFailedNotifyFailed = 70,
+TarsFailedNotifyProcessing = 71,
+TarsFailedNotifyAccepted = 72,
+TarsProcessingNotifyFailed = 73,
+TarsAcceptedNotifyFailed = 74,
+TarsAcceptedNotifyAccepted = 75,
+TarsProcessingNotifyProcessing = 76,
+}
```
Comment (roberthall12) [inline src/functions/retryProcessing/application/__tests__/common/TestEnums.ts:105]: Alignment is off in this file for some reason,

Comment (MarkPatt89) [reply src/functions/retryProcessing/application/__tests__/common/TestEnums.ts:105]: I think it's Github just being silly. Pull the branch and it displays fine.

**Review by MarkPatt89 — APPROVED** [review]
I like this, we should definitely have a session with AT, however, to ensure they're familiar with this.

File: src/functions/retryProcessing/application/__tests__/AutosaveErrorsToRetry.int.ts [hunk 2]
```diff
@@ -18,23 +36,228 @@ describe('AutoSaveErrorsToRetry', () => {
...
+
+const getTestResultData = (): AutosaveTestData[] => {
+  return [
+    {
+      applicationReference: ErrorsToRetryTestCases.TarsFailedRsisFailed,
+      staffNumber: '1',
+      driverSurname: 'Bloggs',
+      resultStatus: ProcessingStatus.PROCESSING,
+      autosave: true,
+    },
```
Comment (roberthall12) [inline src/functions/retryProcessing/application/__tests__/AutosaveErrorsToRetry.int.ts]: Could be refactored into a function, up to you if you think it's worth it:

`function createRecord(appRef: number, staffNum: string, interf: number, uploadStat: number, retryCou: number) {
  return {
    applicationReference: appRef,
    staffNumber: staffNum,
    interface: interf,
    uploadStatus: uploadStat,
    retryCount: retryCou,
  };
}

createRecord(ErrorsToRetryTestCases.TarsFailedRsisFailed, '1', InterfaceIds.TARS, ProcessingStatus.FAILED, 0);`

**Review by roberthall12 — APPROVED** [review]
Nice one!

File: src/functions/retryProcessing/application/__tests__/RetryProcessor.int.ts [hunk 3]
```diff
@@ -182,5 +193,127 @@ describe('RetryProcessor database test', () => {
...
+                               rsisStatus: string,
+                               rsisRetryCount: number,
+                               notifyStatus: string,
+                               notifyRetryCount: number): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      const formattedQuery = db.format(
+            `CALL sp_create_retry_process_scenario(?,?,?,?,?,?,?,?,?)`, [
+              appRef, resultStatus, dateTime, tarsStatus, tarsRetryCount,
+              rsisStatus, rsisRetryCount, notifyStatus, notifyRetryCount]);
+      db.query(
```
Comment (roberthall12) [inline src/functions/retryProcessing/application/__tests__/RetryProcessor.int.ts:220]: I like this, probably worth adding one for calling the autosave procedure too or editing this one to be able to use an autosave flag as the only thing changing will be the procedure name

File: src/functions/retryProcessing/application/__tests__/common/TestEnums.ts (hunk 1 above)
Comment (rossfauldsbjss) [reply src/functions/retryProcessing/application/__tests__/common/TestEnums.ts:105]: Wonder if it's tabs, I'll take a look

File: src/functions/retryProcessing/application/__tests__/CleanupUploadQueueRecords.int.ts [hunk 4]
```diff
@@ -114,56 +120,51 @@ describe('Clean up operations', () => {
...
         retryCount: 0,
       },
       {
         applicationReference: RetryUploadCleanUpTestCases.TarsAcceptedNotifyAcceptedRsisAccepted,
         staffNumber: '1',
         timestamp: '2019-09-01 13:59:59',
-        interface: 1, // NOTIFY
-        uploadStatus: 1, // ACCEPTED
+        interface: InterfaceIds.NOTIFY, // NOTIFY
+        uploadStatus: ProcessingStatus.ACCEPTED, // ACCEPTED
```
Comment (TomBillingtonUK) [inline src/functions/retryProcessing/application/__tests__/CleanupUploadQueueRecords.int.ts]: Now you are using enmus the comments seem a bit pointless

**Review by TomBillingtonUK — APPROVED** [review]
A lot of tests there , but it's definitely easier to understand. Nice one :)

File: src/functions/retryProcessing/application/__tests__/CleanupUploadQueueRecords.int.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/functions/retryProcessing/application/__tests__/CleanupUploadQueueRecords.int.ts]: Definitely - removed them in other files. will remove these.

**Pull request merged**